includes/admin/class-wp-job-manager-promoted-jobs-admin.php

```diff
@@ -114,6 +128,54 @@ public function add_action_notice( $actions_handled ) {
 		return $actions_handled;
 	}
 
+	/**
+	 * Add the host of the Promote Job form to the array of allowed redirect hosts.
+	 *
+	 * @param array $hosts Allowed redirect hosts.
+	 * @return array Updated array of allowed redirect hosts.
+	 */
+	public function add_to_allowed_redirect_hosts( $hosts ) {
+		$hosts[] = wp_parse_url( WP_Job_Manager_Helper_API::get_api_base_url(), PHP_URL_HOST );
+		return $hosts;
+	}
+
+	/**
+	 * Handle the action to promote a job listing, validating as well as redirecting to the form on WPJobManager.com.
+	 *
+	 * @return void
+	 */
+	public function handle_promote_job() {
+		$post_id = absint( $_GET['post_id'] ?? 0 );
+		check_admin_referer( self::PROMOTE_JOB_ACTION . '-' . $post_id );
+		if ( ! $post_id ) {
+			wp_die( esc_html__( 'No job listing ID provided for promotion.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		if ( ! is_ssl() ) {
+			wp_die( esc_html__( 'You must be using SSL to promote a job listing.', 'wp-job-manager' ), '', [ 'back_link' => true ] );
+		}
+		if ( ! current_user_can( 'manage_job_listings', $post_id ) || 'job_listing' !== get_post_type( $post_id ) ) {
```

Comment:
I thought about using `job_manager_user_can_edit_job`, but given that the deactivate action was using this logic, I decided to use the same here.

WDYT? Is there any options you prefer?

Reply:
I had thought of a specific capability, but I think it's fine for now, and we could iterate in the future if needed.

Reply:
@jom, I noticed you got the issue https://github.com/Automattic/WP-Job-Manager/issues/2494. Since @fjorgemota is already implementing it here, we could just hide the buttons as part of that issue?

Reply:
Yeah, I think the only missing piece is hidding the buttons, probably using a similar logic. :D